mentorship ios/Models/HomeModel.swift

```diff
@@ -29,14 +33,154 @@ final class HomeModel: ObservableObject {
+            struct Received: Decodable {
+                let accepted: [RequestStructure]?
+                let rejected: [RequestStructure]?
+                let completed: [RequestStructure]?
+                let cancelled: [RequestStructure]?
+                let pending: [RequestStructure]?
+            }
+        }
+
+        let asMentee: AsMentee?
+        struct AsMentee: Decodable {
+            let sent: Sent?
+            struct Sent: Decodable {
+                let accepted: [RequestStructure]?
+                let rejected: [RequestStructure]?
+                let completed: [RequestStructure]?
+                let cancelled: [RequestStructure]?
+                let pending: [RequestStructure]?
+            }
+            let received: Received?
+            struct Received: Decodable {
+                let accepted: [RequestStructure]?
+                let rejected: [RequestStructure]?
+                let completed: [RequestStructure]?
+                let cancelled: [RequestStructure]?
+                let pending: [RequestStructure]?
+            }
+        }
+
+        enum CodingKeys: String, CodingKey {
+            case asMentor = "as_mentor"
+            case asMentee = "as_mentee"
+        }
+    }
+
+    struct RequestStructure: Decodable {
+        let id: Int?
+        let actionUserID: Int?
+        let mentor: Mentor
+        struct Mentor: Decodable {
+            let id: Int?
+            let userName: String?
+
+            enum CodingKeys: String, CodingKey {
+                case id
+                case userName = "user_name"
+            }
+        }
+        let mentee: Mentee
+        struct Mentee: Decodable {
+            let id: Int?
+            let userName: String?
+
+            enum CodingKeys: String, CodingKey {
+                case id
+                case userName = "user_name"
+            }
+        }
+        let acceptDate: Double?
+        let startDate: Double?
+        let endDate: Double?
+        let notes: String?
+
+        enum CodingKeys: String, CodingKey {
+            case id, mentor, mentee, notes
+            case actionUserID = "action_user_id"
+            case acceptDate = "accept_date"
+            case startDate = "start_date"
+            case endDate = "end_date"
+        }
     }
-    
+
+    struct RelationsListData {
+        let relationTitle = [
+            "Pending Requests",
+            "Accepted Requests",
+            "Rejected Requests",
+            "Cancelled Relations",
+            "Completed Relations"
+        ]
+
+        let relationImageName = [
+            "arrow.2.circlepath.circle.fill",
+            "checkmark.circle.fill",
+            "xmark.circle.fill",
+            "trash.circle.fill",
+            "archivebox.fill"
+        ]
+
+        let relationImageColor: [Color] = [
+            .blue,
+            .green,
+            .pink,
+            .gray,
+            DesignConstants.Colors.defaultIndigoColor
+        ]
+
```

Comment:
Can we put RelationsListData Struct in another file which contain constants, as these are not very much related to HomeModel

Reply:
It is related to HomeModel. It is used to load the relations list in the home screen in the ideal order. The relationCount array is then updated after backend call completes.

Reply:
Then put only relationCount in HomeModel. Icons are not related to model. Right?

Reply:
I've updated the PR. Icons and colors have now been added to respective constants file.

Reply:
That’s good! but I think you misunderstood me. I was talking about shifting whole struct (excluding relationCount) to a new location (most probably a file which contain UI helpers such as colors, images) not just localising names.
As of now we are accessing the icon images and colors from Home model, which is not a good practice. 
Icons and colors acts as Utilities for our UI.

The PR is looking great! Will approves after the following changes

Reply:
Oh, I don't understand why it's a bad practice. Color and images are data only, just the way we have strings and integers in our model, we can have these too. In fact, this is what Apple recommends. 

If I'm still missing your point, please let me know along with the implementation details too.

Reply:
Apple recommends to follow proper architectural design in application. Colors, images and text (pending requests, completed requests, accepted requests, rejected requests) are not the data we are receiving from the backend, these are constants and are used for UI purposes only. I have put a suggestion because I think we can put them in Constants folder under RelationListData file and in Home.swift fetch relationListData like icon/color/name for constants folder. And fetch relationCount from HomeModel.

If you are planning to change the API endpoint from /dashboard to /home, please make change accordingly
Have a great weekend! 😇

Reply:
I don't think that model is only for data coming from the backend. My understanding is that the Home Model should contain data about the home screen. The RelationList struct contains data (text, icons, colors, etc.) that is to be shown in the relations list on the home screen.
As far as I know, Apple too categorizes it all as data. If there's a specific documentation that you're referring to, can you please share its link?

Reply:
Ohh, I think that's a new learning for me. Its just that for future use cases like implementing the rejected and completed sections in my profile view (as discussed) we can fetch them from constants folder rather than HomeModel. (which I think makes sense). 
No problem, I understand what you are saying, current implementation looks fine!
Are you going to use /home api endpoint or should I approve this one?

Reply:
Ohh, I think that's a new learning for me. Its just that for future use cases like implementing the rejected and completed sections in my profile view (as discussed) we can fetch them from constants folder rather than HomeModel. (which I think makes sense). 
No problem, I understand what you are saying, current implementation looks fine!
Are you going to use /home api endpoint or should I approve this one?

Reply:
I'll probably use the home api. But, that's more suited for the backend consumption PR I guess.